qiskit/aqua/algorithms/phase_estimators/phase_estimator.py

```diff
@@ -0,0 +1,213 @@
+"""The Quantum Phase Estimation Algorithm."""
+
+from typing import Dict, Optional, Union
+from qiskit.circuit import QuantumCircuit
+import qiskit
+#from qiskit.result import Counts
+from qiskit.circuit.library import PhaseEstimation
+from qiskit.circuit.classicalregister import ClassicalRegister
+from qiskit.providers import BaseBackend
+from qiskit.aqua import QuantumInstance
+from qiskit.aqua.algorithms import QuantumAlgorithm
+from qiskit import execute
+from qiskit.aqua.utils import get_subsystem_density_matrix
+import numpy as np
+
+
+class PhaseEstimator(QuantumAlgorithm):
+    """The Quantum Phase Estimation algorithm.
```

Comment:
At some point we should add a description of the algorithm and references here (not pressing but just so its written down somewhere)

Reply:
We also need a very precise description of the requirements that the inputs must satisfy. For example, a particular structure is assumed for the input QPE circuit.